com.chartboost.helium/Android~/helium-android-bridge/src/main/java/com/chartboost/heliumsdk/unity/HeliumUnityBridge.kt

```diff
@@ -0,0 +1,387 @@
+import com.chartboost.heliumsdk.unity.HeliumEventProcessor.serializeHeliumRewardEvent
+import com.chartboost.heliumsdk.unity.HeliumEventProcessor.serializePlacementIlrdData
+import com.chartboost.heliumsdk.unity.HeliumUnityAdWrapper.Companion.wrap
+import com.unity3d.player.UnityPlayer
+
+@Suppress("NAME_SHADOWING")
+class HeliumUnityBridge {
+    private var lifeCycleEventListener: ILifeCycleEventListener? = null
+    private var bannerEventsListener: IBannerEventListener? = null
+    private var interstitialEventsListener: IInterstitialEventListener? = null
+    private var rewardedEventListener: IRewardedEventListener? = null
+    private var ilrdObserver: HeliumIlrdObserver? = null
+
+    fun setupEventListeners(
+        lifeCycleListener: ILifeCycleEventListener,
+        interstitialListener: IInterstitialEventListener,
+        rewardedListener: IRewardedEventListener,
+        bannerListener: IBannerEventListener
+    ) {
+        lifeCycleEventListener = lifeCycleListener
+        interstitialEventsListener = interstitialListener
+        rewardedEventListener = rewardedListener
+        bannerEventsListener = bannerListener
+    }
+
+    fun setSubjectToCoppa(isSubject: Boolean) {
+        HeliumSdk.setSubjectToCoppa(isSubject)
+    }
+
+    fun setSubjectToGDPR(isSubject: Boolean) {
+        HeliumSdk.setSubjectToGDPR(isSubject)
+    }
+
+    fun setCCPAConsent(hasGivenConsent: Boolean) {
+        HeliumSdk.setCCPAConsent(hasGivenConsent)
+    }
+
+    fun setUserHasGivenConsent(hasGivenConsent: Boolean) {
+        HeliumSdk.setUserHasGivenConsent(hasGivenConsent)
+    }
+
+    fun onBackPressed(): Boolean {
+        return HeliumSdk.onBackPressed()
+    }
+
+    var userIdentifier: String?
+        get() = HeliumSdk.getUserIdentifier()
+        set(userIdentifier) {
+            HeliumSdk.setUserIdentifier(userIdentifier)
+        }
+
+    fun start(appId: String, appSignature: String, unityVersion: String) {
+        ilrdObserver = object : HeliumIlrdObserver {
+            override fun onImpression(impData: HeliumImpressionData) {
+                val json = serializePlacementIlrdData(impData.placementId, impData.ilrdInfo)
+                lifeCycleEventListener?.DidReceiveILRD(json)
+            }
+        }
+
+        runTaskOnUiThread {
+            // This call initializes the Helium SDK. This might change in the future with two ID parameters and we'll get rid of the logControllerListener
+            HeliumSdk.start(UnityPlayer.currentActivity, appId, appSignature) { error: Error? ->
+                if (error == null) {
+                    Log.d("Unity", "HeliumUnityBridge: Plugin Initialized")
+                    HeliumSdk.setGameEngine("unity", unityVersion)
+                    ilrdObserver?.let {
+                        HeliumSdk.subscribeIlrd(it)
+                    }
+                } else {
+                    Log.d("Unity", "HeliumUnityBridge: Plugin failed to initialize: $error")
+                }
```

Comment:
```kt
error?.let {
    Log.d("Unity", "HeliumUnityBridge: Plugin failed to initialize: $error")
} ?: run {
    Log.d("Unity", "HeliumUnityBridge: Plugin Initialized")
    HeliumSdk.setGameEngine("unity", unityVersion)
    ilrdObserver?.let {
        HeliumSdk.subscribeIlrd(it)
    }
}

```

Reply:
This is awesome :D i will add it, it didn't occur to me this was an alternative, but also it's a lot of legacy code so I have not revised a lot of things.